### PR TITLE
Get rooms

### PR DESCRIPTION
**Pull request by JohnOldman2018**

### CODE REVIEW

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.jantoleu.rooms_manager.controllers;
```
Comment (amussagaliyev) [inline src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: We already have RoomController. So please use it for controller methods related to Room

File: src/main/java/com/jantoleu/rooms_manager/model/Room.java [hunk 2]
```diff
@@ -14,4 +14,10 @@
     private String name;
     private String countryCode;
     private Boolean lightIsOn;
+
+
+
+    public Boolean getLightIsOn () {
```
Comment (amussagaliyev) [inline src/main/java/com/jantoleu/rooms_manager/model/Room.java]: Why do we need this getter? We already have lombok generating all necessary getters and setters. Please remove

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import com.jantoleu.rooms_manager.model.Room;
+import com.jantoleu.rooms_manager.service.RoomService;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.websocket.server.PathParam;
+import java.util.List;
+
+@PostMapping("/api/room/{roomId}/{status:on/off}")
```
Comment (amussagaliyev) [inline src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: We agreed on different url pattern.
It should be /api/room/{roomId}
Other parameters should come within request body

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+public class RoomSwitch {
+
+    private RoomService roomService;
+    List<Room> rooms = roomService.getAllRooms();
```
Comment (amussagaliyev) [inline src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: Don't use global variables

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    public Boolean setRoomById (@PathParam(value="light_is_on") Boolean switchParam, Model model, List<Room> rooms) {
+        model.addAttribute("nameAttribute", switchParam);
+        for (Room rm: rooms) {
```
Comment (amussagaliyev) [inline src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: You are retrieved all rooms.
Then you are traversing through all rooms.
I assume you wanted to find the room by ID and then change its light_is_on attribute but you don't.
Too much logic for a controller. Controller should call service. Service should call Repository. I don't see it.

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.List;
+
+@PostMapping("/api/room/{roomId}/{status:on/off}")
+public class RoomSwitch {
+
+    private RoomService roomService;
+    List<Room> rooms = roomService.getAllRooms();
+
+    public Boolean setRoomById (@PathParam(value="light_is_on") Boolean switchParam, Model model, List<Room> rooms) {
+        model.addAttribute("nameAttribute", switchParam);
```
Comment (amussagaliyev) [inline src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: You don't have to deal with model at all.
This endpoint does not return page.
This is API endpoint which performs its business logic and returns some response in JSON format not a model or page.

**Review by amussagaliyev — CHANGES_REQUESTED** [review]

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java (hunk 5 above)
Comment (JohnOldman2018) [reply src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: I see.

File: src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java (hunk 1 above)
Comment (JohnOldman2018) [reply src/main/java/com/jantoleu/rooms_manager/controllers/RoomSwitch.java]: ok

**Pull request merged**